addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -209,6 +209,10 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
             const options = await this._getAddProductOptions(product);
             // Do not proceed on adding the product when no options is returned.
             // This is consistent with _clickProduct.
+            if (product.barcode_packaging_qty) {
+                Object.assign(options, { quantity: product.barcode_packaging_qty });
+                delete product.barcode_packaging_qty;
+            }
```

Comment:
At this point `options` can be `undefined`, therefore, this should be put after the check below (the `if (!options) return;`).